WeakAuras/Types.lua

```diff
@@ -1495,24 +1504,23 @@ for id, str in pairs(Private.combatlog_spell_school_types) do
   Private.combatlog_spell_school_types_for_ui[id] = ("%.3d - %s"):format(id, str)
 end
 
----@type fun(): number
----@field GetCurrencyIDFromLink fun(currencyLink: string): number
----@field ExpandCurrencyList fun(index: number, expand: boolean): any
----@field GetCurrencyListInfo fun(index: number): table
 if WeakAuras.IsRetail() then
   Private.GetCurrencyListSize = C_CurrencyInfo.GetCurrencyListSize
   Private.GetCurrencyIDFromLink = C_CurrencyInfo.GetCurrencyIDFromLink
   Private.ExpandCurrencyList = C_CurrencyInfo.ExpandCurrencyList
   Private.GetCurrencyListInfo = C_CurrencyInfo.GetCurrencyListInfo
 elseif WeakAuras.IsWrathOrCata() then
   Private.GetCurrencyListSize = GetCurrencyListSize
+  ---@type fun(currencyLink: string): number?
   Private.GetCurrencyIDFromLink = function(currencyLink)
     local currencyID = string.match(currencyLink, "|Hcurrency:(%d+):")
-    return currencyID
+    return tonumber(currencyID)
```

Comment:
@Boneshockx can you verify this is correct? It looks like it's correct (C_CurrencyInfo.GetCurrencyIDFromLink returns number, not string)? You touched this code last, so you own it :).

Reply:
Yeah tonumber should be correct here.